### DISCUSSION TITLE
Data structure for multiple datasets?

### DISCUSSION THREAD

**Discussion by jwbrooks0** [pydata/xarray#6164] [Q&A] [answered]
**Question**
Does xarray support a data structure for multiple datasets?  Or is there a native python (like lists or dicts) or a 3rd party solution?  I presently address this need by storing datasets in different HDF5 groups (but within the same hdf5 file) but would prefer an option to keep the data in memory.  Thanks!  

**Background**
I often find myself with data from the same experiment that I wish to keep "together" but are also somewhat incompatible.  As an example, several measurements may have a time coordinate, t = np.arange(0, 1e6, 1e-7) and others will have a time coordinate, t = np.arange(1e8, 1e-10).  I supposed one solution could be to rename the second time coordinate (e.g. "t2" instead of "t") so that all the data fits into the same Dataset, but this will cause some issues with some of my adjacent code.  

**Misc**
I love xarray.  I've been using it for the past few years and spread the word whenever I can.  Please keep up the good work.

**Comment by keewis** [comment, marked as answer]
you should be able to use the experimental [`DataTree`](https://github.com/TomNicholas/datatree) for this.

**Reply by jwbrooks0** [reply]
That appears to be exactly what I'm looking for.  Thank you very much!

**Comment by Illviljan** [comment]
> one solution could be to rename the second time coordinate (e.g. "t2" instead of "t")

This is the approach I've been taking, I use the group name though, `time_group0`, so it's easier to know where it came from.
